sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -186,7 +186,13 @@ def __call__(
             work_group_size=work_group_size,
         )
 
-        X_t = dpctl.tensor.from_numpy(X.T, device=self.device)
+        # The kernels expect X to be shaped as (n_features, n_samples) instead
+        # of the usual (n_samples, n_features). Furthermore, using a Fortran
+        # memory layout is actually more L1 cache friendly on the GPU and
+        # results in a slight performance increase (typically 15% faster) when
+        # updating the centroids from the assigned samples.
```

Comment:
I'm not sold on the explanation of the results we've seen yet, can we phrase this more than as a suggestion of what we think happens rather than a definitive explanation ? in particular as I've understood it, it goes against what the paper suggests. Also, before setting confident defaults we would need testing on a larger range of hardware.